src/gsy_e/models/leaves.py

```diff
@@ -61,7 +76,14 @@ class Leaf(Area):
     strategy_type = None
 
     def __init__(self, name, config, uuid=None, **kwargs):
-        if config.external_connection_enabled:
+        if ConstSettings.MASettings.MARKET_TYPE == SpotMarketTypeEnum.COEFFICIENTS.value:
+            # For the SCM only use the SCM-enabled strategies and ignore the rest.
+            try:
+                self.strategy_type = scm_strategy_mapping[self.strategy_type]
+            except KeyError as e:
+                logging.error(f"Strategy {self.strategy_type} not supported in SCM.")
```

Comment:
Nit: this will raise a warning because of the lazy formatting.
```suggestion
                logging.error("Strategy %s not supported in SCM.", self.strategy_type)
```

Reply:
Fixed, same for the other pylint warnings in leaves.py